chucks-pick3-web/angular-app/src/app/components/sign-up.component/sign-up.component.ts

```diff
@@ -1,19 +1,47 @@
 import {Component, OnInit} from '@angular/core';
-
+import {FormValidationService} from '../../services/form-validation.service/form-validation.service';
+import {MemberService} from '../../services/member.service/member.service';
 @Component({
   selector: 'app-sign-up',
   templateUrl: './sign-up.component.html',
   styleUrls: ['./sign-up.component.css']
 })
 export class SignUpComponent implements OnInit {
+  // Todo: create an errorMessage attribute so that it can be use later if and when the memberService.createAccount() return false.
+ public setErrorMessage(querySelector, errorMessage, input) {
+    document.querySelector(querySelector).innerHTML = errorMessage;
+    input.setCustomValidity(errorMessage);
+
+  }
 
-  constructor() {}
+  // Todo: inject both the FormValidationService & MemberService to be able to delegate to later.
+  constructor(public formValidationService: FormValidationService ,
+              private memberService: MemberService ) {
+  }
 
   ngOnInit() {
     const howToActive = document.getElementById('howTo');
     howToActive.classList.remove('active');
     howToActive.classList.add('allow-hover');
   }
+  public onSubmit(input) {
+   const isFormValid =  this.formValidationService.validateForm();
+   if (isFormValid === true) {
+     isFormValid.memberService.createAccount();
+   } else if (isFormValid === false) {
+     this.setErrorMessage(".invalid-feedback." + input.name, '', input);
+   }
+
+  }
+
+  // Todo: create onSubmit() method that delegates call to:
```

Comment:
Remove comments and todos in all TS files.

Reply:
Removed all comments and todos.